drivers/ethernet/eth_stm32_hal.c

```diff
@@ -117,24 +247,57 @@ static int eth_tx(struct device *dev, struct net_pkt *pkt)
 		goto error;
 	}
 
-	dma_tx_desc = heth->TxDesc;
-	while ((dma_tx_desc->Status & ETH_DMATXDESC_OWN) != (uint32_t)RESET) {
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+	const uint32_t u32_cur_tx_desc = 0;  /* heth->TxDescList.CurTxDesc; */
+#endif
+
+	dma_tx_desc = GET_FIRST_DMA_TX_DESC(heth);
+	while (IS_ETH_DMATXDESC_OWN(dma_tx_desc) != (uint32_t)RESET) {
```

Comment:
Please move existing code modification to dedicated commits

Reply:
Comes with next push.